### ISSUE TITLE
Removing the `subdir` and `tag` options with `git`/`hg` command

### ISSUE THREAD

**Issue by tobil4sk** [HaxeFoundation/haxelib#538]
A while ago I noticed that the `subdir` and `tag`/`revision` features weren't documented so I created a PR to do that #531. I didn't notice however that the reason for this was because some years ago they were intended to be removed:
- 8ee54470ba5544f5f263f03aece7cd7a918868e9
- https://github.com/HaxeFoundation/haxelib/tree/vcs_version_remove

The reason behind removing `subdir` is explained in this issue: #238 (starting at https://github.com/HaxeFoundation/haxelib/issues/238#issuecomment-193492391)

I also found this to be somewhat relevant: #263. The feature is a bit of a relic.

The only time I have ever used this feature is for installing the [hashlink haxelib](https://lib.haxe.org/p/hashlink), which tends to be out of date. However, there are other solutions to that problem, such as separating it into a separate repository.

Is it ok to continue with their deprecation and removal? Having to accommodate for these features has always been a bit of an inconvenience, and they are not even supported in `haxelib.json` dependencies so removing them shouldn't break too much. The only potential hurdle is that both of them are positional, which could be an issue if either should be kept. However, I would argue that if that were the case they should be replaced with command specific option flags: #319.

If it is ok to remove it then I'm happy to take it on myself after #510.

**Comment by tobil4sk** [comment]
> The subdir option is useful for a lot of hashlink libraries (hlsdl, hlopenal, hldx, hashlink) and it's been around for a long time so I think it would be better not to remove it.

https://github.com/HaxeFoundation/haxelib/issues/263#issuecomment-1256955617

However, I still think it would be good to stop using positional arguments and instead use `--subdir` etc. for these options.

**Comment by tobil4sk** [comment]
Closing in favour of #319
